src/datamodel_code_generator/__main__.py

```
@@ -1172,6 +1190,250 @@ def _try_rebuild_model(obj: type) -> None:
 
 
+def _get_base_model_parents(model_class: type) -> list[type]:
+    """Get parent classes that are BaseModel subclasses (excluding BaseModel itself)."""
+    return [
+        p
+        for p in model_class.__bases__
+        if isinstance(p, type) and issubclass(p, BaseModel) and p is not BaseModel
+    ]
+
+
+def _transform_single_model_to_inheritance(  # noqa: PLR0912
```

Ruff (0.14.10)

Unused `noqa` directive (non-enabled: `PLR0912`)

Remove unused `noqa` directive

(RUF100)